src/Thrift.Net.Compiler/Program.cs

```diff
@@ -58,50 +58,62 @@ public class Program
+                    outputDirectory);
 
-            using (var stream = input.OpenRead())
-            {
-                var result = Compiler.Compile(stream);
+                using (var stream = input.OpenRead())
+                {
+                    var result = Compiler.Compile(stream);
 
-                OutputCompilationSummary(console, result);
+                    OutputCompilationSummary(console, result);
 
-                // TODO: Pull message formatting out to its own object.
-                foreach (var message in result.Messages.OrderBy(message => message.LineNumber))
-                {
-                    console.Out.Write($"{thriftFile.RelativePath}({message.LineNumber},{message.StartPosition}-{message.EndPosition}): {message.MessageType} {message.FormattedMessageId}: {message.Message} [{input.FullName}]{Environment.NewLine}");
-                }
+                    // TODO: Pull message formatting out to its own object.
+                    foreach (var message in result.Messages.OrderBy(message => message.LineNumber))
+                    {
+                        console.Out.Write($"{thriftFile.RelativePath}({message.LineNumber},{message.StartPosition}-{message.EndPosition}): {message.MessageType} {message.FormattedMessageId}: {message.Message} [{input.FullName}]{Environment.NewLine}");
+                    }
 
-                if (result.HasErrors)
-                {
-                    return (int)ExitCode.CompilationFailed;
-                }
+                    if (result.HasErrors)
+                    {
+                        return (int)ExitCode.CompilationFailed;
+                    }
 
-                if (result.Document.ContainsDefinitions)
-                {
-                    var generatedCode = DocumentGenerator.Generate(result.Document);
+                    if (result.Document.ContainsDefinitions)
+                    {
+                        var generatedCode = DocumentGenerator.Generate(result.Document);
 
-                    FileWriter.Write(thriftFile, generatedCode);
+                        FileWriter.Write(thriftFile, generatedCode);
+                    }
                 }
-            }
 
-            return (int)ExitCode.Success;
+                return (int)ExitCode.Success;
+            }
+            catch (Exception exception)
+            {
+                console.Out.Write($"Error: {exception}{Environment.NewLine}{Environment.NewLine}");
+                console.Out.Write($"Thrift.Net has encountered an error. This is a problem with the compiler and not caused by your code.{Environment.NewLine}");
+                console.Out.Write($"Please help us to resolve this by reporting a new issue here: https://github.com/adamconnelly/Thrift.Net/issues/new?template=issue_template.md. {Environment.NewLine} {Environment.NewLine}");
+                console.Out.Write($"We welcome contributions, so please feel free to create a PR to resolve the issue.{Environment.NewLine}");
```

Comment:
```suggestion
                console.Out.Write($"We welcome contributions, so feel free to create a PR to resolve the issue.{Environment.NewLine}");
```

After reading it I just figure we had too many "pleases" in there :smile_cat: